batch7crud-roll5/src/main/java/com/lftechnology/batch7crud/controller/EmployeeController.java

```diff
@@ -0,0 +1,167 @@
+
+import com.lftechnology.batch7crud.model.Employee;
+import com.lftechnology.batch7crud.services.EmployeeService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet implementation class EmployeeController
+ */
+
+@WebServlet("/employees/*")
+public class EmployeeController extends HttpServlet {
+    EmployeeService employeeService = new EmployeeService();
+    Employee emp = new Employee();
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * @see HttpServlet#HttpServlet()
+     */
+    public EmployeeController() {
+        super();
+        // TODO Auto-generated constructor stub
+    }
+
+    /**
+     * @see HttpServlet#doGet(HttpServletRequest request, HttpServletResponse response)
+     */
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        // TODO Auto-generated method stub
+        String path = request.getPathInfo();
+        System.out.println(path);
+        if (path != null) {
+
+            String[] pathParts = path.split("/");
+            for (String s : pathParts) {
+                System.out.println("parts: " + s);
+            }
+            if (pathParts[1].equals("create"))
+                create(request, response);
+
+            else if (pathParts[2].equals("edit")) {
+                int id = Integer.parseInt(pathParts[1]);
+                edit(request, response, id);
+            }
+
+        }
+
+        else {
+            fetchData(request, response);
+        }
+
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        // TODO Auto-generated method stub
+        // doGet(request, response);
+        String path = request.getPathInfo();
+        if (path != null) {
+            System.out.println(path);
+            String[] pathParts = path.split("/");
+            if (pathParts[1].equals("createProcess")) {
+                createProcess(request, response);
+            } else if (pathParts[2].equals("editProcess")) {
+                int id = Integer.parseInt(pathParts[1]);
+                editProcess(request, response, id);
+            }
+        }
+
+    }
+
+    private void edit(HttpServletRequest request, HttpServletResponse response, int id) {
+        // TODO Auto-generated method stub
+        try {
+            emp = employeeService.fetchById(id);
+            request.setAttribute("employee", emp);
+            RequestDispatcher dispatcher = request.getRequestDispatcher("/WEB-INF/views/editEmployee.jsp");
+            dispatcher.forward(request, response);
+
+        } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)